code/src/main/webapp/JavaScript/script.js

```diff
@@ -0,0 +1,28 @@
+class LandingPage {
+  constructor(){
+    this.background = document.getElementById("wrapper-background");
+  }
+
+  hide() {
+    this.background.classList.remove("landing-background");
+  }
+}
+
+class LoginPage {
+  constructor(){
+    this.background = document.getElementById("wrapper-background");
+  }
+
+  show() {
+    this.background.classList.add("login-background");
+  }
+}
+
+function TransitionFromLandingToLogin() {
```

Comment:
Great work on the two classes. To switch between them let's create another class:

class PageController {
  constructor () {
    this.landingPage = new LandingPage();
    this.loginPage = new LoginPage();
    this.currentlyShown = "none";
  }

  show(pageToShow) {
    // Put the history.pushState here!
    this.hideAllPages();
    if (pageToShow === "login") {
      this.loginPage.show();
    }
    ...
  }
}

cosnt PAGE_CONTROLLER = new PageController();
PAGE_CONTROLLER.show("landing");